### PR TITLE
All claims 42 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds a page for homeless / at risk of becoming homeless questions to the 526 All Claims form.

## Testing done
- Tested locally
- Added unit tests for required functionality

## Screenshots
Basic screen when a veteran hits this page (selecting 'no' means no additional questions asked):
screen shot 2018-09-17 at 1 43 53 pm

-----

If a veteran selects the 'I’m currently homeless' option, additional questions are expanded:
screen shot 2018-09-17 at 1 44 21 pm

-----

If a veteran then responds with 'Other' for their current homeless housing situation, we expand a follow-up question asking them to describe:
screen shot 2018-09-17 at 1 44 36 pm

-----

Optionally, a veteran can enter a contact name / phone number if they're homeless or at risk of becoming homeless. If a veteran fills out either one of these fields, the other field becomes required (this is what's required by the EVSS endpoint):
screen shot 2018-09-17 at 1 44 50 pm

-----

If a veteran indicates that they're 'at risk of becoming homeless', we expand a different set of questions:
screen shot 2018-09-17 at 1 45 02 pm

If a veteran then indicates that their living situation is 'other', we then also ask them to describe their situation just like we do with 'currently homeless' responses:
screen shot 2018-09-17 at 1 45 11 pm

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/config/schema.js [hunk 1]
```diff
@@ -1045,6 +1045,54 @@ const schema = {
...
+      'enum': [
+        'No',
+        'homeless',
+        'atRisk'
+      ]
+    },
+    homelessHousingSituation: {
+      type: 'string',
+      'enum': [
+        'I’m living in a homeless shelter.',
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/config/schema.js]: Do we think we'll end up with codes to send to EVSS for this? I'm a little wary of descriptive strings in `enum` fields like this.

File: src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        no: 'No',
+        homeless: homelessLabel,
+        atRisk: atRiskLabel
+      }
+    }
+  },
+  'view:isHomeless': {
+    'ui:options': {
+      expandUnder: 'homelessOrAtRisk',
+      expandUnderCondition: (housing) => housing === 'homeless'
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js]: I think you can put `'homeless'` here instead of a whole function.

**Review by jbalboni — APPROVED** [review]
Couple comments, but LGTM.

**Review by peggygannon — CHANGES_REQUESTED** [review]
@markgreenburg See below for some content edits: 

Please update these bullets to: [making contractions and adding periods] 
- I'm **currently** homeless.
- I'm **at risk** of becoming homeless.

Please capitalize Yes No under this question: 
Do you need to quickly leave your current living situation?
- Yes
- No

Please tweak contact label slightly TO:
Please provide the name and number of a person we can call if we need to get in touch with you.

Name of person we can contact

File: src/applications/disability-benefits/all-claims/config/schema.js (hunk 1 above)
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/config/schema.js]: I think we will need to send them specific values for this, but it's not clear what that'll be yet.

File: src/applications/disability-benefits/all-claims/tests/pages/homelessOrAtRisk.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    const form = mount(
+      <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{
+          homelessOrAtRisk: 'homeless',
+          'view:isHomeless': {
+            homelessHousingSituation: 'Other',
+            otherHomelessHousing: 'An airplane',
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/pages/homelessOrAtRisk.unit.spec.jsx]: :thinking:

File: src/applications/disability-benefits/all-claims/tests/pages/homelessOrAtRisk.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  it('should submit when user not homeless or at risk', () => {
+    const onSubmit = sinon.spy();
+
+    const form = mount(
+      <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{
+          homelessOrAtRisk: 'No'
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/pages/homelessOrAtRisk.unit.spec.jsx]: Should there be constants for these?

File: src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    },
+    homelessHousingSituation: {
+      'ui:title': 'Please describe your current living situation.',
+      'ui:required': (formData) => _.get('homelessOrAtRisk', formData, '') === 'homeless',
+      'ui:widget': 'radio'
+    },
+    otherHomelessHousing: {
+      'ui:title': 'Please describe',
+      'ui:required': (formData) => _.get('view:isHomeless.homelessHousingSituation', formData, '') === 'Other',
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js]: Should there be constants for these?

File: src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    }
+  },
+  'view:isHomeless': {
+    'ui:options': {
+      expandUnder: 'homelessOrAtRisk',
+      expandUnderCondition: (housing) => housing === 'homeless'
+    },
+    homelessHousingSituation: {
+      'ui:title': 'Please describe your current living situation.',
+      'ui:required': (formData) => _.get('homelessOrAtRisk', formData, '') === 'homeless',
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/homelessOrAtRisk.js]: Should there be constants for these?

**Review by cvalarida — APPROVED** [review]
Looks good, but I think we can reduce the number of magic strings.

File: src/applications/disability-benefits/all-claims/config/schema.js (hunk 1 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/config/schema.js]: Right...I should probably switch to at least using labels for the full text and tightening up the enums.

**Comment by markgreenburg** [comment]
@peggygannon here's a screenshot with all the new content:

screen shot 2018-09-18 at 10 55 50 am

**Review by peggygannon — APPROVED** [review]

**Comment by peggygannon** [comment]
Thanks, @markgreenburg. Updated screenshots look good.

**Pull request merged**